views/js/tools/upload/tester.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2015 (original work) Open Assessment Technologies SA ;
+ */
+define([
+    'jquery',
+    'lodash',
+    'async',
+    'util/url'
+], function($, _, async, urlHelper) {
+    'use strict';
+
+    /**
+     * A binary kilo bytes (KiB)
+     * @type {Number}
+     * @private
+     */
+    var _kilo = 1024;
+
+    /**
+     * A binary mega bytes (MiB)
+     * @type {Number}
+     * @private
+     */
+    var _mega = _kilo * _kilo;
+
+    /**
+     * Result of calibration requests
+     */
+    var data = [];
```

Comment:
Be careful with module variables. Here we cannot launch multiple tests at the same time because the same dataset is used by the tester. And once the check is finished, the data remains and will pollute the next run.

You should rather put this var either as a parameter or as a closure inside the tester factory and provide the result with the promise it returns. That way you will ensure to start from a clean and secure context each time.